mvapich/default.nix

```diff
@@ -23,15 +42,35 @@ in stdenv.mkDerivation rec {
   ];
 
 
-  nativeBuildInputs = [ pkgconfig bison ];
+  nativeBuildInputs = [ pkgconfig bison makeWrapper ];
   propagatedBuildInputs = [ numactl rdma-core zlib infiniband-diags opensm ];
-  buildInputs = [ numactl libxml2
-                  perl gfortran
-                  slurm openssh
-                  hwloc rdma-core
-                  infiniband-diags opensm ];
+  buildInputs = [
+    numactl
+    libxml2
+    perl
+    gfortran
+    slurm
+    openssh
+    hwloc
+    rdma-core
```

Comment:
It would make sense to also make the packages for respective interfaces conditional depending on which interface type is chosen. That may avoid that configure picks up unnecessary packages.